src/coreclr/src/jit/codegenlinear.cpp

```diff
@@ -1446,6 +1446,15 @@ void CodeGen::genConsumeRegs(GenTree* tree)
         }
 #endif // FEATURE_HW_INTRINSICS
 #endif // TARGET_XARCH
+#if defined(FEATURE_HW_INTRINSICS) && defined(TARGET_ARMARCH)
+        else if (tree->OperIs(GT_HWINTRINSIC))
+        {
+            const HWIntrinsic hwIntrin(tree->AsHWIntrinsic());
+
+            assert((hwIntrin.id == NI_Vector64_GetElement) || (hwIntrin.id == NI_Vector128_GetElement));
+            genConsumeReg(tree);
```

Comment:
I have a question - this is only needed because now one HWIntrinsic node can contain another HWIntrinsic node?

Reply:
> Shouldn't this be consuming the operands of tree rather than tree itself?

I tried to mimic the behavior of scenarios when the node was not marked contained earlier. In that case, we would have called [genConsumeReg(tree)](https://github.com/dotnet/runtime/blob/master/src/coreclr/src/jit/codegenlinear.cpp#L1466). In fact I changed it to `tree` from `tree->gtGetOp1()` in https://github.com/dotnet/runtime/pull/37139/commits/48f09b7bd4dc1e0e9c9188c0e57ae102e18a0a15 when I realized this.

> this is only needed because now one HWIntrinsic node can contain another HWIntrinsic node?

Correct.